File: depend/depend-web-servlet/src/main/java/net/yan100/compose/webservlet/properties/ServletWebApplicationProperties.java
package net.yan100.compose.webservlet.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.http.converter.StringHttpMessageConverter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * spring web mvc http servlet 配置属性
 *
 * @author TrueNine
 * @since 2023-02-20
 */
@Data
@ConfigurationProperties(prefix = "compose.web-servlet")
public class ServletWebApplicationProperties {
  List<String> allowConverters = new ArrayList<>(Arrays.asList("getDocumentation",
    "swaggerResources",
    "openapiJson"));
  List<Class<?>> allowConverterClasses = new ArrayList<>(List.of(StringHttpMessageConverter.class));
}

File: depend/servlet/src/main/kotlin/net/yan100/compose/depend/servlet/properties/ServletWebApplicationProperties.kt
package net.yan100.compose.depend.servlet.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.http.converter.StringHttpMessageConverter
import kotlin.reflect.KClass

private const val PREFIX = "compose.depend.servlet"

/**
 * spring web mvc http servlet 配置属性
 *
 * @author TrueNine
 * @since 2023-02-20
 */
@ConfigurationProperties(prefix = PREFIX)
data class ServletWebApplicationProperties(
  var allowConverters: MutableList<String> =
    mutableListOf("getDocumentation", "swaggerResources", "openapiJson"),
  var allowConverterClasses: MutableList<KClass<*>> =
    mutableListOf(StringHttpMessageConverter::class),
)
